sprint_2

Ilya-andreevi4 opened the pull request.

Ilya-andreevi4 committed:
modified:   package.json

Ilya-andreevi4 committed:
Add ts and lints. Moved to hbs-precompile.

Ilya-andreevi4 committed:
add routing and orginaze code

Ilya-andreevi4 committed:
Осталось валидация и работа с запросами

Ilya-andreevi4 committed:
Доделал валидацию форм

Ilya-andreevi4 committed:
Доработал messageInput + addFilesPopup

Ilya-andreevi4 committed:
Добавил class для работы с запросами.

Ilya-andreevi4 committed:
Fix regexp

mikebars1995 reviewed (commented):
К сожалению, работу придется отклонить без потери итерации, так как сейчас невозможно проверить работоспособность проекта:

* Проект не устанавливается https://skr.sh/sIdTSMuC5NI
* По ссылке на задеплоеный проект на нетлифай находится версия проекта из 1го спринта

Ilya-andreevi4 committed:
node fix

mikebars1995 commented on src/utils/page-fields.ts:
```diff
@@ -0,0 +1,132 @@
+};
+
+const SECOND_NAME = {
+  id: "last_name",
```
**Нужно исправить**


По заданию 1го спринта нужно давать инпутам **конкретные** атрибуты `name`, чтобы в 3м спринте не менять ничего при подключении к `Api`. Скрин из задания https://disk.yandex.ru/i/lSPqeGrAa20ayQ

mikebars1995 commented on src/pages/login-page/index.ts:
```diff
@@ -0,0 +1,62 @@
+        submitButton: new Button({
+          label: "Enter",
+          className: "modal",
+        }),
```
**Можно лучше**


Лучше давать разный `type` для разных кнопок

Например, если это кнопка сабмита, то у нее должен быть `type = "submit"`, иначе, не будет работать встроенная логика обработки формы по нажатию на `Enter`

mikebars1995 commented on src/utils/fetch.ts:
```diff
@@ -0,0 +1,142 @@
+  return `${!prefix && "?"}${res.join("&")}`;
+}
+
+type HTTPMethod = (
```
Отлично, что не дублируете типизацию аргументов

mikebars1995 commented on package.json:
```diff
@@ -1,28 +1,35 @@
-    "build": "parcel build index.hbs",
-    "start": "node server.js"
+    "build": "parcel build static/index.html",
+    "start": "parcel build static/index.html && node server.js",
```
**Можно лучше**

Чтобы **автоматически запускался** проект в браузере, нужно дописать в скрипт `--open`

Ilya-andreevi4 replied:
После того как добавил name для inputs и type для buttons долго не мог разобраться с ошибками при деплое на netlify. так что очень много пришлось сделать правок, надо было изначально это все делать в ветке для деплоя. Прошу прощения за спам коммитами

mikebars1995 reviewed (commented):
Здравствуйте. (**Нужно развернуть общий комментарий ↓**)

Работа проделана огромная


Что сделано отлично:


* `Readme` хорошо оформлен
* Проект задеплоен и отлично работает
* Отлично, что не забыли про `.gitignore`
* Отлично, что ограничиваете версию `node`
* Отлично, что фиксируете версии зависимостей
* Отлично, что нет `EOF` ошибок в гите
* Хорошая структура папок и файлов
* Добавлен `ESLint` и `Stylelint`
* Отлично, что цвета вынесены в `css`-константы
* Отлично, что не забываете про `alt` в тегах `img`
* Отлично, что не забываете про тег `main` для главной части страницы
* Хорошая реализация общего класса `Block`
* Отлично, что устанавливаете обработчики через объект `events` в пропсах
* Отлично, что моковые данные вынесены в отдельный файл

Что нужно исправить:

* По заданию 1го спринта нужно давать инпутам **конкретные** атрибуты `name`, чтобы в 3м спринте не менять ничего при подключении к `Api`. Скрин из задания https://disk.yandex.ru/i/lSPqeGrAa20ayQ 

Что можно улучшить:

* Лучше давать разный `type` для разных кнопок
* Чтобы **автоматически запускался** проект в браузере, нужно дописать в скрипт `--open`


Исправьте, пожалуйста, недочеты и работа будет принята. Пожалуйста, проверьте работоспособность проекта и наличие возможных ошибок в консоли браузера (кнопка `F12`) перед отправкой на ревью. 

Напоминаю, что работа может быть принята только после исправления всех критических замечаний `Нужно исправить`.

**Комментарии `Можно лучше` не обязательны к исправлению прямо сейчас, это рекомендации**

Удачного рефакторинга кода.

Ilya-andreevi4 committed:
add name to input and change type for buttons

Ilya-andreevi4 committed:
renamed:    static/media/7334443_3594626.jpg:Zone.Identifier -> "static/media/7334443_3594626.jpg\357\200\272Zone.Identifier"

Ilya-andreevi4 committed:
modified:   README.md
	modified:   package-lock.json
	deleted:    "static/media/7334443_3594626.jpg\357\200\272Zone.Identifier"

Ilya-andreevi4 committed:
fix commit

Ilya-andreevi4 committed:
modified:   package-lock.json
	modified:   package.json
	modified:   src/utils/page-fields.ts
	modified:   tsconfig.json

Ilya-andreevi4 committed:
fix

Ilya-andreevi4 committed:
Trouble with Netlify

Ilya-andreevi4 committed:
Merge branch 'sprint_2'

Ilya-andreevi4 committed:
modified:   netlify.toml
	modified:   tsconfig.json

Ilya-andreevi4 committed:
fix deploy

Ilya-andreevi4 committed:
fix deploy

Ilya-andreevi4 committed:
fix deploy

Ilya-andreevi4 committed:
fix deploy

Ilya-andreevi4 committed:
fix deploy

Ilya-andreevi4 committed:
fix deploy

Ilya-andreevi4 committed:
fix conflict package.json

mikebars1995 reviewed (commented):
Поздравляю! Ваша работа принята.

Вы отлично потрудились.

Удачного дальнейшего обучения.

Ilya-andreevi4 committed:
Merge branch 'main' into sprint_2
